modules/healthcheck/default.nix

```
@@ -0,0 +1,281 @@
+              message = "When healthcheck.readiness-probe is enabled, you must define `healthcheck.exec` with the service command.";
+            }) (lib.attrNames servicesWithHealthcheck)
+          );
+
+          systemd = {
+            timers = lib.mapAttrs' (
+              mainServiceName: serviceConfig:
+              let
+                cfg = serviceConfig.healthcheck;
+              in
+              {
+                name = "${mainServiceName}-liveness-check";
+                value = mkIf (cfg != null && cfg.liveness-probe.enable) (
+                  let
+                    probe = cfg.liveness-probe;
+                  in
+                  {
+                    description = "Timer for ${mainServiceName} liveness probe";
+                    timerConfig = {
+                      Unit = "${mainServiceName}-liveness-check.service";
+                    };
+                    wantedBy = [ "${mainServiceName}.service" ];
+                  }
+                );
+              }
+            ) (servicesWithHealthcheck);
+
+            services =
+              (lib.mapAttrs (
+                mainServiceName: serviceConfig:
+                let
+                  cfg = serviceConfig.healthcheck;
+                in
+                (mkIf (cfg.readiness-probe.enable) (
+                  let
+                    probeCfg = cfg.readiness-probe;
+                  in
+                  {
+                    serviceConfig.Type = lib.mkForce "notify";
+                    serviceConfig.TimeoutStartSec = lib.mkForce "infinity";
+
+                    # 2. Build the wrapper script that becomes the new ExecStart.
+                    serviceConfig.ExecStart =
+                      let
+                        scriptPath = makeBinPath (
+                          [
+                            pkgs.systemd
+                            pkgs.curl
+                            pkgs.gawk
+                          ]
+                          ++ (if cfg ? "path" then cfg.path else [ ])
+                          ++ (if serviceConfig ? "path" then serviceConfig.path else [ ])
+                        );
+                        successStatus = builtins.replaceStrings [ "%OUTPUT%" ] [ ''''${output}'' ] probeCfg.statusReady;
+                      in
+                      lib.mkForce (
+                        pkgs.writeShellScript "${mainServiceName}-readiness-check" ''
+                          #!${pkgs.runtimeShell}
+                          # set -o errexit
+                          set -o nounset
+                          export PATH="${scriptPath}:$PATH"
+
+                          check() {
+                            echo "Health check: starting background readiness probe for ${mainServiceName}."
+                            while true; do
+                              local output
+                              output=$(sh -c "${probeCfg.command}" 2>/dev/null)
+
+                              if [[ -n "''${output}" ]]; then
```

I don't think the output of the probe command is useful, so I suggest simplifying the code to only check the return code:

```suggestion
                              if ${probeCfg.command}; then
```